src/hover_provider/modelHoverProvider.ts

```diff
@@ -0,0 +1,160 @@
+          const mdString = this.getHoverMarkdownFor(
+            project.getProjectName(),
+            dbtModel[0],
+            document.uri,
+          );
+          if (mdString !== undefined) {
+            const hover = new Hover(mdString, new Range(position, position));
+            resolve(hover);
+          }
+          this.telemetry.sendTelemetryEvent("provideModelHover", {
+            type: "single",
+          });
+          return;
+        }
+        if (dbtModel && dbtModel.length === 3) {
+          const mdString = this.getHoverMarkdownFor(
+            dbtModel[0],
+            dbtModel[2],
+            document.uri,
+          );
+          if (mdString !== undefined) {
+            const hover = new Hover(mdString, new Range(position, position));
+            resolve(hover);
+          }
+          this.telemetry.sendTelemetryEvent("provideModelHover", {
+            type: "dual",
+          });
+          return;
+        }
+      }
+      resolve(undefined);
+    });
+  }
+
+  private onManifestCacheChanged(event: ManifestCacheChangedEvent): void {
+    event.added?.forEach((added) => {
+      this.modelToLocationMap.set(added.projectRoot.fsPath, added.nodeMetaMap);
+    });
+    event.removed?.forEach((removed) => {
+      this.modelToLocationMap.delete(removed.projectRoot.fsPath);
+    });
+  }
+
+  private getHoverMarkdownFor(
+    projectName: string,
+    modelName: string,
+    currentFilePath: Uri,
+  ): MarkdownString | undefined {
+    const projectRootpath =
+      this.dbtProjectContainer.getProjectRootpath(currentFilePath);
+    if (projectRootpath === undefined) {
+      return;
+    }
+    const nodeMap = this.modelToLocationMap.get(projectRootpath.fsPath);
+    if (nodeMap === undefined) {
+      return;
+    }
+    const node = nodeMap.get(modelName);
+    if (node) {
+      const content = new MarkdownString();
+      content.supportHtml = true;
+      content.isTrusted = true;
+      content.appendMarkdown(
+        `<span style="color:#347890;">(ref)&nbsp;</span><span><strong>${node.alias}</strong></span>`,
+      );
+      if (node.description !== null) {
+        content.appendMarkdown(`</br><span>${node.description}</span>`);
+      }
+      content.appendText("\n");
+      content.appendText("\n");
+      content.appendMarkdown("---");
+      content.appendText("\n");
+      content.appendText("\n");
+      for (const colKey in node.columns) {
+        const column = node.columns[colKey];
+        content.appendMarkdown(
+          `<br/><span style="color:#347890;">(column)&nbsp;</span><span>${column.name} &nbsp;</span>`,
+        );
+        if (column.data_type !== null) {
+          content.appendMarkdown(
+            `<span>-&nbsp;${column.data_type.toUpperCase()}</span>`,
+          );
+        }
+        if (column.description !== null) {
+          content.appendMarkdown(
+            `<br/><span><em>${column.description}</em></span>`,
+          );
+        }
+      }
```

Comment:
When no description is provided, the spacing is a bit extraneous. Could you reduce it so it has the same spacing compared to the case when description is there.

Reply:
I thought I had handled that in a recent commit but I was checking for `null` instead of empty strings in the descriptions. Looks a lot better without the spacing now.
![Screenshot 2023-09-19 at 7 26 43 PM](https://github.com/AltimateAI/vscode-dbt-power-user/assets/79863718/fc2bad0e-6437-4c06-a6f6-d9d0ae72cf34)
![Screenshot 2023-09-19 at 7 27 20 PM](https://github.com/AltimateAI/vscode-dbt-power-user/assets/79863718/749860ed-9ec5-4705-898c-e202aed141c6)